src/cqrs/Aggregate.ts

```diff
@@ -13,43 +13,64 @@ export class Aggregate {
     const aggregate = Aggregate.instantiateAggregate(firstEventExecutor.getAggregateType(), firstEventExecutor.getAggregateId(firstEvent), aggregateClasses);
     allEvents.forEach((e) => {
       const eventExecutor = EventExecutor.getEventExecutor(e, eventExecutors);
-      if (eventExecutor) {
-        eventExecutor.apply(e, aggregate);
-      } else {
-        throw new Error(`Unknown event during getAggregate: ${firstEvent.constructor.name}`);
-      }
+      Aggregate.applyEventOnAggregate(e, eventExecutor, aggregate);
     });
     return aggregate;
   }
 
+  public static applyEventOnAggregate(e: Object, eventExecutor: EventExecutor<any, any>, aggregate: Aggregate) {
+    if (eventExecutor) {
+      if (aggregate.eventApplied(eventExecutor.getEventId(e))) {
```

Comment:
Use the ordinal to check if the event has been applied, not the list of event ids